### PR TITLE
Cleaned up version of earlier PR #48601.

### PR DESCRIPTION
**Pull request by PeterSolMS**
Port of ConserveMemory setting we implemented on .NET Framework.

### CODE REVIEW

File: src/coreclr/gc/gc.cpp [hunk 1]
```diff
@@ -18285,7 +18298,38 @@ int gc_heap::joined_generation_to_condemn (BOOL should_evaluate_elevation,
             n = max_generation;
             *blocking_collection_p = TRUE;
             settings.loh_compaction = TRUE;
-            dprintf (GTC_LOG, ("compacting LOH due to hard limit"));
+            dprintf (GTC_LOG, ("compacting gen2+loh due to hard limit"));
+        }
+    }
+
+    if ((conserve_mem_setting != 0) && (n >= max_generation))
```
Comment (Maoni0) [inline src/coreclr/gc/gc.cpp:18305]: n cannot be > max_generation... it can only be at most max_generation.

File: src/coreclr/gc/gc.cpp [hunk 2]
```diff
@@ -23360,6 +23404,21 @@ size_t gc_heap::get_total_gen_estimated_reclaim (int gen_number)
...
 }
 
+size_t gc_heap::get_total_gen_size (int gen_number)
+{
+#ifdef MULTIPLE_HEAPS
+    size_t size = 0;
+    for (int hn = 0; hn < gc_heap::n_heaps; hn++)
+    {
+        gc_heap* hp = gc_heap::g_heaps[hn];
+        size += hp->generation_sizes (hp->generation_of (gen_number));
```
Comment (Maoni0) [inline src/coreclr/gc/gc.cpp:23414]: this should use `generation_size`, not `generation_sizes`

File: src/coreclr/gc/gc.cpp (hunk 1 above)
Comment (PeterSolMS) [reply src/coreclr/gc/gc.cpp:18305]: I know - I changed the >= to ==.

File: src/coreclr/gc/gc.cpp (hunk 2 above)
Comment (PeterSolMS) [reply src/coreclr/gc/gc.cpp:23414]: Yup, thanks for catching!

**Review by Maoni0 — APPROVED** [review]
LGTM!

**Pull request merged**